content/departments/product-engineering/engineering/enablement/dev-experience/newsletter.md

```diff
@@ -8,6 +8,73 @@ To learn more about components of Sourcegraph's developer experience, check out
+
+It is the responsibility of authors of recently failed builds to investigate what might have gone wrong, seek help if needed, and help get the pipeline back green. We hope this will prevent long periods of time where many commits to `main` go untested due to failing jobs. To learn more, check out the [branch lock playbook](../../process/incidents/playbooks/ci.md#buildchecker-has-locked-the-main-branch)
+
+We've also made significant investments towards improving and streamlining the pipeline for better stability and observability - most recently, [a large number of E2E/QA tests were dropped](https://github.com/sourcegraph/sourcegraph/pull/28995) - which will hopefully help with minimizing locks triggered by test and infrastructure flakes.
+
+### Sentry integration to monitor internal pipeline scripts and hooks
+
+There are scripts and components of the CI pipeline that should never fail, independently of the tests results. These have proved be to hard to monitor, especially when the scripts are called from build hooks. Being notified when these failures happen enables faster reaction time. Here is an [example](https://github.com/sourcegraph/sourcegraph/pull/28915/files#diff-3c4244f37fc751696252758dd92a887d9e1e30851b18932c142ae56202bb5ea7R40) to get monitor a command so that a Sentry issue in the [Buildkite](https://sentry.io/organizations/sourcegraph/projects/buildkite/?project=6110304) project is created on a non zero exit code.
+
+### Specifying tools and language versions ran by _any_ continuous pipeline
+
+In response to [INC-59](https://docs.google.com/document/d/1HXKZa9L3MVswK6pDpRN5TdCgUcEcQca9Re4vCwlb6Ek/edit#) we have reworked which tools and languages versions are to be used in a given CI job. Previously, the agents where running a mix of `asdf` and natively installed versions which created trouble when diagnosing build failures that weren't caused by the test themselves. 
+
+It is now _the responsibility of each repository to provide an adequate `.tools-version` file that defines what are the versions it needs_. There are no more pre-installed `go` version for example. 
+Presently, this approach is limited by having the plugin for that particular tool installed beforehand on the agents images (we are working on removing this limitation). The overarching goal is to make the agents reasonably independent from what they are actually building. 
+
+### Observability
```

Comment:
@bobheadxi should we be more precise regarding which observability we're talking about?

Reply:
What are the options for observability? :o

Reply:
That wasn't very clear, sorry about that. I mean: should we group topics by their big themes like, observability, CI stability, etc. or instead name this section `# Grafana templates updates` and the following one `# Revamped quickstart docs for local development`?

Reply:
Oh yeah now that you've added a bunch of topics the themes are a lot clearer now, I've shuffled things around into sections!

Reply:
Awesome thank you!